### PR TITLE
feat: export vega-expression and vega-event-selector

### LINKED ISSUE #2917: Export vega-expression and vega-event-selector
Both are used by Vega-Lite and exporting them would allow Vega-Lite to ship with a smaller bundle.

### PR DESCRIPTION
**Pull request by domoritz**
fixes #2917

### CODE REVIEW

**Comment by domoritz** [comment]
Needed for https://github.com/vega/vega-lite/pull/7263

**Review by jheer — APPROVED** [review]
LGTM

**Comment by domoritz** [comment]
One thing I wonder is whether it is better to export with a different name or export the exact same name a package exports. The advantage of the latter is that one can do a rename in rollup as we do in https://github.com/vega/vega-lite/blob/e19d9ac23396dc9b3579279f4f66a37173ffa098/rollup.config.js#L34. 

Do use the renamed exports (as in this pull request), I had to change Vega-Lite do depend on Vega directly (as opposed to specific packages). I don't think it's a big issue for Vega-Lite since Vega is already a peer dependency but I wanted to see what you think before we go down this route.

**Comment by jheer** [comment]
Sometimes it might be necessary to rename an export -- obviously if there is a name collision, but also if the name becomes to ambiguous in the enlarged context. So I'm not averse to that. That said, if the name works well enough on its own, my bias would be to prefer keeping names unchanged. In this case, the expression exports need to be renamed (e.g., we can't have two _parse_ methods), but I'm not sure the selector export needs to be renamed.

File: packages/vega-typings/types/runtime/index.d.ts [hunk 1]
```diff
@@ -257,3 +257,7 @@ export * from 'vega-util';
 export * from './dataflow';
 export * from './renderer';
 export * from './scene';
+
+export { codegen as expressionCodegen, parse as expressionParse } from 'vega-expression';
+
+export { selector as parseSelector } from 'vega-event-selector';
```
Comment (arvind) [inline packages/vega-typings/types/runtime/index.d.ts:263]: Would be good to keep the naming scheme parallel. So parseSelector/Expression or selector/expressionParse (I prefer the former).

Comment (jheer) [reply packages/vega-typings/types/runtime/index.d.ts:263]: I similarly prefer `parseExpression` over `expressionParse`.

Comment (domoritz) [reply packages/vega-typings/types/runtime/index.d.ts:263]: Btw, we use `runtimeContext` above. Do you also prefer `codegenExpression` (it sounds less natural than `expressionCodegen` as we go from expression AST to code)?

Comment (jheer) [reply packages/vega-typings/types/runtime/index.d.ts:263]: I'm fine with `codegenExpression` for parity with `parseExpression`

Comment (domoritz) [reply packages/vega-typings/types/runtime/index.d.ts:263]: Done

File: packages/vega-typings/types/runtime/index.d.ts [hunk 2]
```diff
@@ -260,4 +260,4 @@ export * from './scene';
 
 export { codegen as codegenExpression, parse as parseExpression } from 'vega-expression';
 
-export { selector as parseSelector } from 'vega-event-selector';
```
Comment (arvind) [inline packages/vega-typings/types/runtime/index.d.ts]: I think my prev comment was unclear — I had wanted us to be consistent (`parseSelector` and `parseExpression` or `selectorParse` and `expressionParse`). I think we should keep the rename here as `parseSelector`.

Comment (domoritz) [reply packages/vega-typings/types/runtime/index.d.ts]: Done

**Review by arvind — APPROVED** [review]

**Pull request merged**